### PR TITLE
#2231 Broken head in syntax

### PR DESCRIPTION
**Pull request by maxonfjvipon**
Closes: #2231

---

### Detailed summary

- Deleted `catches-same-line-name.yamleo-parser/src/test/resources/org/eolang/parser/typos/scoped-application-after-suffix.yaml`
- Added `first` and `a > x (b > x)` in `eo` field of `eo-parser/src/test/resources/org/eolang/parser/typos/scoped-application-after-suffix.yaml`
- Changed `count(error)` from `1` to `2` in `eo-parser/src/test/java/org/eolang/parser/SyntaxTest.java`
- Deleted `skip: true` in `eo-parser/src/test/resources/org/eolang/parser/typos/broken-head.yaml`
- Changed regex pattern in `eo-parser/src/main/java/org/eolang/parser/RedundantParentheses.java`
- Added `# Comment().` in `eo-parser/src/test/java/org/eolang/parser/RedundantParenthesesTest.java`
- Changed grammar rules in `eo-parser/src/main/antlr4/org/eolang/parser/Program.g4`
- Added methods `exitAbstraction` and `enterType` in `eo-parser/src/main/java/org/eolang/parser/XeListener.java`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by maxonfjvipon** [comment]
@yegor256 please have a look

**Comment by yegor256** [comment]
@volodya-lombrozo pls, take a look

File: eo-parser/src/test/java/org/eolang/parser/RedundantParenthesesTest.java [hunk 1]
```diff
@@ -96,7 +96,8 @@ static Stream<Arguments> testCases() {
             Arguments.of("[] > foo\n  ((1.add 1)) > x", false),
             Arguments.of("[] > foo\n  1.add 1 > x\n  (1.add 1)", false),
             Arguments.of("[] > foo\n  1.add 1 > x\n  (1.add 1) > y", false),
-            Arguments.of("[] > foo\n  add > x\n    (1)\n    (1)\n", false)
+            Arguments.of("[] > foo\n  add > x\n    (1)\n    (1)\n", false),
+            Arguments.of("# Comment().\n[] > obj\n", true)
```
Comment (volodya-lombrozo) [inline eo-parser/src/test/java/org/eolang/parser/RedundantParenthesesTest.java:100]: @maxonfjvipon As I understand, the `correct` examples go first before the `incorrect` examples:
```
true
true
...
false,
false
```
You have:
```
true
true
...
false,
false,
true
```
Maybe it's better to move the last statement higher?

**Review by volodya-lombrozo — CHANGES_REQUESTED** [review]

File: eo-parser/src/test/java/org/eolang/parser/RedundantParenthesesTest.java (hunk 1 above)
Comment (maxonfjvipon) [reply eo-parser/src/test/java/org/eolang/parser/RedundantParenthesesTest.java:100]: @volodya-lombrozo sure

**Comment by maxonfjvipon** [comment]
@volodya-lombrozo please have a look one more time

**Review by volodya-lombrozo — APPROVED** [review]
@maxonfjvipon Looks good to me, thanks.
@yegor256 Could you merge that changes, please?

**Comment by yegor256** [comment]
@rultor merge

**Comment by maxonfjvipon** [comment]
@yegor256 please try to merge one more time

**Comment by yegor256** [comment]
@rultor merge

**Pull request merged**